### PR TITLE
SPM-1185: Implement the order_query reflection attribute

### PR DESCRIPTION
**Pull request by semtexzv**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Review by Josca — CHANGES_REQUESTED** [review]
Some test error:
~~~
 test         | manager/controllers/filter_test.go:56:48: cannot use dummyParser (value of type func(v string) (interface{}, error)) as string value in struct literal (typecheck)
test         | 		attrMap := database.AttrMap{"test": {"test", dummyParser}}
test         | 		                                             ^
test         | manager/controllers/filter_test.go:76:54: cannot use dummyParser (value of type func(v string) (interface{}, error)) as string value in struct literal (typecheck)
test         | 		attrMap := database.AttrMap{"test": {"(NOT test)", dummyParser}}
~~~

And pls include SPM-1185, thx

File: base/database/query.go [hunk 1]
```diff
@@ -15,8 +15,9 @@ type AttrParser func(string) (interface{}, error)
 
 type AttrName = string
 type AttrInfo struct {
-	Query  string
-	Parser AttrParser
+	DataQuery  string
```
Comment (Josca) [inline base/database/query.go:18]: For historic reasons I would keep just `Query` here. So we don't need to update all "query" annotations.

Comment (semtexzv) [reply base/database/query.go:18]: It's npt changed in the annotations, just clarifies it in the reflection code.

**Review by Josca — CHANGES_REQUESTED** [review]
Test fail:
~~~
test         | === RUN   TestOrderQuery
test         |     query_test.go:83: 
test         |         	Error Trace:	query_test.go:83
test         |         	Error:      	Not equal: 
test         |         	            	expected: ""
test         |         	            	actual  : "REVERSE(am.text_note)"
test         |         	            	
test         |         	            	Diff:
test         |         	            	--- Expected
test         |         	            	+++ Actual
test         |         	            	@@ -1 +1 @@
test         |         	            	-
test         |         	            	+REVERSE(am.text_note)
test         |         	Test:       	TestOrderQuery
test         | --- FAIL: TestOrderQuery (0.00s)
~~~

**Review by Josca — APPROVED** [review]

**Pull request merged**